R/data-predictor.R

```diff
@@ -663,11 +664,77 @@ data_ac <- function(bterms, data, data2, basis = NULL, ...) {
   if (use_ac_cov_time(acef)) {
     # data for the 'covariance' versions of time-series structures
     out$N_tg <- length(unique(tgroup))
-    out$begin_tg <- as.array(ulapply(unique(tgroup), match, tgroup))
-    out$nobs_tg <- as.array(with(out,
-      c(if (N_tg > 1L) begin_tg[2:N_tg], N + 1) - begin_tg
-    ))
-    out$end_tg <- with(out, begin_tg + nobs_tg - 1)
+    # if using latent variables, add grouping levels
+    if (has_ac_latent_residuals(bterms)) {
```

Comment:
this looks good to me at first glance, but I would like to review this in a call with you.